### PR TITLE
More helm

### LINKED ISSUE #2263: Result of ListApps operation for Git repo is cached incorrectly
Result of `ListApps` operation for Git repo is cached incorrectly. Cache key should use resolved revision, not ambiguous:

https://github.com/argoproj/argo-cd/blob/master/reposerver/repository/repository.go#L84-L84

Helm repo is also cached incorrectly. ListApps does not use revision at all, so it should not be used in cache key.

### LINKED ISSUE #2269: Path autocompletion is broken on application create panel
**Describe the bug**

~~Looks like the happy path is broken:~~ Path autocompletion is broken on application create panel.

**To Reproduce**

1. Try to create any app using Argo CD UI and Git repo.
2. Make sure repo is not cloned into temp directory.
3. ~~Repo server keep complaining that repo does not have app~~ See path field has no suggestions

### PR DESCRIPTION
**Pull request by alexec**
Closes #2263.
Closes #2269.

### CODE REVIEW

File: reposerver/repository/repository.go [hunk 1]
```diff
@@ -81,17 +81,17 @@ func (s *Service) ListApps(ctx context.Context, q *apiclient.ListAppsRequest) (*
 	if err != nil {
 		return nil, err
 	}
-	if apps, err := s.cache.ListApps(q.Repo.Repo, q.Revision); err == nil {
-		log.Infof("cache hit: %s/%s", q.Repo.Repo, q.Revision)
-		return &apiclient.AppList{Apps: apps}, nil
-	}
 	apps, resolvedRevision, err := r.ListApps(q.Revision)
```
Comment (alexmt) [inline reposerver/repository/repository.go]: In this case, we are not taking any advantage of caching. We need to resolve revision before locking the repository. Check the cache and then lock repo, fetch, checkout, search for apps etc.

Same is in `GetRevisionMetadata`

**Review by alexmt — CHANGES_REQUESTED** [review]

File: ui/src/app/settings/components/repos-list/repos-list.tsx [hunk 2]
```diff
@@ -161,7 +161,7 @@ export class ReposList extends React.Component<RouteComponentProps<any>> {
                                     <FormField formApi={formApi} label='Skip server verification' field='insecure' component={CheckboxField}/>
                                 </div>
                                 <div className='argo-form-row'>
-                                    <FormField formApi={formApi} label='Enable LFS support' field='enableLfs' component={CheckboxField}/>
+                                    <FormField formApi={formApi} label='Enable LFS support (Git only)' field='enableLfs' component={CheckboxField}/>
```
Comment (alexmt) [inline ui/src/app/settings/components/repos-list/repos-list.tsx]: I think we agreed to show fields based on selected type. So `Enable LFS support` should be hidden unless Git is selected

**Review by alexmt — CHANGES_REQUESTED** [review]

File: reposerver/repository/repository.go (hunk 1 above)
Comment (alexec) [reply reposerver/repository/repository.go]: I think we need to lock the repo on any operation. Let me have a deep think

File: ui/src/app/settings/components/repos-list/repos-list.tsx (hunk 2 above)
Comment (alexec) [reply ui/src/app/settings/components/repos-list/repos-list.tsx]: Hai. This PR is a dumping ground for any Helm polish.

File: reposerver/repository/repository.go (hunk 1 above)
Comment (alexmt) [reply reposerver/repository/repository.go]: We use `go-git` to execute `ls-remote` in memory. This allows to check if operation result is available in cache without locking the repository.

Comment (alexmt) [reply reposerver/repository/repository.go]: I did not explicitly mention another issue here. Looks like we execute `ListApps` before checking if result is available in cache or not. That is what I mean by `we are not taking any advantage of caching`

Comment (alexec) [reply reposerver/repository/repository.go]: I've reverted the change, as the fix is more involved. I'll create another issuep

File: ui/src/app/settings/components/repos-list/repos-list.tsx (hunk 2 above)
Comment (alexec) [reply ui/src/app/settings/components/repos-list/repos-list.tsx]: How about labelling it with "(Git only)" for the moment?

Comment (alexmt) [reply ui/src/app/settings/components/repos-list/repos-list.tsx]: sure.

**Review by alexmt — APPROVED** [review]

**Pull request merged**